sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -761,84 +787,204 @@ public T next()
+        // The start of each interval should be aligned with the start of the bucket, and the end interval should be
+        // aligned with the start of the next interval.
+        if (!granularity.bucketStart(intervalStart).equals(intervalStart) || !granularity.bucketStart(intervalEnd).equals(intervalEnd)) {
+          throw new ValidationException("DELETE WHERE clause contains an interval which is not aligned with PARTITIONED BY granularity");
+        }
+      }
+      return intervals
+          .stream()
+          .map(AbstractInterval::toString)
+          .collect(Collectors.toList());
+    }
+
+    private static DimFilter convertQueryToDimFilter(SqlNode replaceTimeQuery, DateTimeZone dateTimeZone)
+        throws ValidationException
+    {
+      if (replaceTimeQuery instanceof SqlBasicCall) {
+        SqlBasicCall sqlBasicCall = (SqlBasicCall) replaceTimeQuery;
+        Pair<String, String> columnValuePair;
+        switch (sqlBasicCall.getOperator().getName()) {
+          case "AND":
+            List<DimFilter> dimFilters = new ArrayList<>();
+            for (SqlNode sqlNode : sqlBasicCall.getOperandList()) {
+              dimFilters.add(convertQueryToDimFilter(sqlNode, dateTimeZone));
+            }
+            return new AndDimFilter(dimFilters);
+          case "OR":
+            dimFilters = new ArrayList<>();
+            for (SqlNode sqlNode : sqlBasicCall.getOperandList()) {
+              dimFilters.add(convertQueryToDimFilter(sqlNode, dateTimeZone));
+            }
+            return new OrDimFilter(dimFilters);
+          case "NOT":
+            return new NotDimFilter(convertQueryToDimFilter(sqlBasicCall.getOperandList().get(0), dateTimeZone));
+          case ">=":
+            columnValuePair = createColumnValuePair(sqlBasicCall.getOperandList(), dateTimeZone);
+            return new BoundDimFilter(columnValuePair.left, columnValuePair.right, null, false, null, null, null, StringComparators.NUMERIC);
+          case "<=":
+            columnValuePair = createColumnValuePair(sqlBasicCall.getOperandList(), dateTimeZone);
+            return new BoundDimFilter(columnValuePair.left, null, columnValuePair.right, null, false, null, null, StringComparators.NUMERIC);
+          case ">":
+            columnValuePair = createColumnValuePair(sqlBasicCall.getOperandList(), dateTimeZone);
+            return new BoundDimFilter(columnValuePair.left, columnValuePair.right, null, true, null, null, null, StringComparators.NUMERIC);
+          case "<":
+            columnValuePair = createColumnValuePair(sqlBasicCall.getOperandList(), dateTimeZone);
+            return new BoundDimFilter(columnValuePair.left, null, columnValuePair.right, null, true, null, null, StringComparators.NUMERIC);
+          default:
+            throw new ValidationException("Unsupported operation in DELETE WHERE clause: " + sqlBasicCall.getOperator().getName());
+        }
+      }
+      throw new ValidationException("Invalid DELETE WHERE clause");
```

Comment:
Instead of ValidationException, does `ParseException` be more appropriate? (`ParseException` that gets autogenerated)

Reply:
If the difference isn't too much, can it be kept as ValidationException?
The parent function already throws SqlParseException, so a different ParseException for this one case might be making things more confusing.

Reply:
SGTM 👍